### PR TITLE
Show inactive users in activity log and team table

### PR DESCRIPTION
**Pull request by mddilley**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/9673

This PR fixes a bug where inactivated users were showing as "Unknown User" in the team table and the activity log. The "Unknown User" came from two separate components and was implemented in two different ways so I pulled out some helpers to align how we are getting user full names and initials from the Hasura response or Local Storage and added some tests.

Let me know if some of these changes are not a good path forward. I think the two helpers cover our bases but I could also be looking at it naively. 🙏 

## Testing
**URL to test:** 
https://deploy-preview-721--atd-moped-main.netlify.app/moped/projects/1859?tab=summary

**Steps to test:**

_The user in these steps is actually a soft deleted user account record with first name of **Inactive** and last name **User**. I just realized this is a little confusing since we have fallback text that says **Unknown User**_

1. Go to the above link and look at the **Team** tab
2. You will see me and "Inactive User" in the table - "Inactive User" is soft deleted and should be italicized and have " - Inactive" on the end
3. Make sure that you cannot select "Inactive User" from the dropdown you use to add a new team member
4. Go to the activity log and you should see a change row where "Inactive User" was added to the team
5. Check the following locations to make sure that user initials still show as expected:
   - Avatar in top right of app (if you don't have an uploaded profile image)
   - Avatar on the account page (if you don't have an uploaded profile image)

There is still one bug where **Unknown User** shows in the **User** column of the **Activity feed** which will be fixed in https://github.com/cityofaustin/atd-data-tech/issues/9423

Screenshot of an inactive user in the Team table
Screen Shot 2022-07-18 at 10 12 21 AM

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectActivityLog.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectActivityLog.js:142]: I refactored this so that we go `moped_user` ▶️ initials instead of `moped_user` ▶️ full name ▶️ initials. It is moved to `userNames.js` further up in the diff. Maybe there is a better names for that file?

File: moped-editor/src/utils/userNames.js
Comment (mddilley) [inline moped-editor/src/utils/userNames.js:2]: Not sure this will ever really happen since we validate first name and last name are present on user creation but it could be good to be ready for abnormal data if it were to ever happen.

**Review by frankhereford — APPROVED** [review]
Thanks for the note saying the user account being demo'd in your test was named "Inactive User"! That was puzzling for a second.

I think this looks great, thank you @mddilley!

**Review by roseeichelmann — APPROVED** [review]
Tried this and it all looks good on my end!

File: moped-editor/src/utils/test/userNames.test.js
Comment (chiaberry) [inline moped-editor/src/utils/test/userNames.test.js:16]: ❤️

**Review by chiaberry — APPROVED** [review]

**Comment by mddilley** [comment]
@amenity @patrickm02L I have a few approvals here so this should be ready for PM review when there is time. Happy to update how the inactive user rows look in the Team table. That can be expanded beyond the italicized name style, but I wanted to start simple. 🙏

**Comment by patrickm02L** [comment]
**Cannot delete users from the Project Team**

I tried inactivating a user in order to see if they would show up as inactive on Project Team, they did - I used the Test Testing user to do this. 

Then I reactivated Test Testing and issued a new password in order to see if that changed them back to active on the Project Team, it did. 

But now I can't delete Test Testing from the Project Team. I added Rebecca to see if the problem with deleting was specific to Test Testing because I inactivate/activated them, but I was unable to delete her from the Team either. 


https://user-images.githubusercontent.com/5566727/180237194-c0234610-7263-4014-b72a-a6e1dd679d92.mp4

**Comment by amenity** [comment]
Nice catch, @patrickm02L! Since I'm out until Tuesday, I'll leave it to you to ✔️ **Product manager approved**. 🙏🏻

File: moped-editor/src/views/projects/projectView/ProjectTeamTable.js
Comment (mddilley) [reply moped-editor/src/views/projects/projectView/ProjectTeamTable.js:85]: Filtering the soft deleted project personnel could happen in the query too so let me know if anyone has a preference. 🙏

**Comment by mddilley** [comment]
@patrickm02L I fixed up the bug that you found. Thanks again for uncovering it! 🙏 🐛 🕵️  We should be able to add/remove project personnel and see an accurate list in the table.

**Review by patrickm02L — APPROVED** [review]
Works like a dream! 🤘 

https://user-images.githubusercontent.com/5566727/180326615-1626e5ee-9d82-421e-b281-ecd6deb8776c.mp4

**Comment by patrickm02L** [comment]
I also checked out this the following steps from your PR instructions, and everything worked as expected. 

> 3. Make sure that you cannot select "Inactive User" from the dropdown you use to add a new team member
> 4. Go to the activity log and you should see a change row where "Inactive User" was added to the team
> 5. Check the following locations to make sure that user initials still show as expected

**Pull request merged**
